/*
 * Copyright 2021 Google LLC.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

package yggdrasil_decision_forests.model.cart.proto;

import "yggdrasil_decision_forests/learner/abstract_learner.proto";
import "yggdrasil_decision_forests/learner/decision_tree/decision_tree.proto";

// Training configuration for the CART algorithm.
message CartTrainingConfig {
  // Next ID: 3

  // Decision tree specific parameters.
  optional decision_tree.proto.DecisionTreeTrainingConfig decision_tree = 1;

  // Ratio of the training dataset used to create the validation dataset used
  // to prune the tree.
  optional float validation_ratio = 2 [default = 0.1];
}

extend model.proto.TrainingConfig {
  optional CartTrainingConfig cart_config = 1012;
}
